src/features/views/components/ViewDataTable/columnTypes/PersonTagColumnType/index.tsx

```diff
@@ -0,0 +1,270 @@
+import { AppDispatch, RootState } from 'core/store';
+import { PersonTagViewColumn, ZetkinViewRow } from '../../../types';
+import { tagLoad, tagLoaded } from 'features/tags/store';
+
+type PersonTagViewCell = null | {
+  value?: string;
+};
+
+export default class PersonTagColumnType implements IColumnType {
+  cellToString(cell: PersonTagViewCell): string {
+    return cell?.value ? cell.value.toString() : Boolean(cell).toString();
+  }
+
+  getColDef(
+    column: PersonTagViewColumn,
+    accessLevel: ZetkinObjectAccess['level'],
+    state: RootState,
+    apiClient: IApiClient,
+    dispatch: AppDispatch,
+    orgId: number
+  ): Omit<GridColDef, 'field'> {
+    const tagId = column.config.tag_id;
+
+    const tagList = state.tags.tagList;
+    const tagItem = tagList.items.find((item) => item.id == tagId);
+
+    const tagFuture = loadItemIfNecessary(tagItem, dispatch, {
+      actionOnLoad: () => tagLoad(tagId),
+      actionOnSuccess: (tag) => tagLoaded(tag),
+      loader: () =>
+        apiClient.get<ZetkinTag>(`/api/orgs/${orgId}/people/tags/${tagId}`),
+    });
+
+    const tag = tagFuture.data;
+
+    return {
+      align: 'center',
+      editable: !accessLevel && tag?.value_type !== null,
+      headerAlign: 'center',
+      renderCell: (params: GridRenderCellParams<ZetkinViewRow, ZetkinTag>) => (
+        <Cell cellValue={params.value} personId={params.row.id} tag={tag} />
+      ),
+      renderEditCell: (params) => (
+        <ValueTagEditCell tagColor={tag?.color} {...params} />
+      ),
+      sortComparator: (v1: ZetkinTag, v2: ZetkinTag) => {
+        return compareTags(v1, v2);
+      },
+    };
+  }
+
+  getSearchableStrings(cell: PersonTagViewCell): string[] {
+    return cell?.value ? [cell.value] : [];
+  }
+
+  handleKeyDown(
+    viewGrid: UseViewGridReturn,
+    column: PersonTagViewColumn,
+    personId: number,
+    data: PersonTagViewCell,
+    ev: MuiEvent<KeyboardEvent<HTMLElement>>,
+    accessLevel: ZetkinObjectAccess['level']
+  ): void {
+    if (accessLevel) {
+      // Any non-null value means we're in restricted mode
+      return;
+    }
+
+    if (ev.key == 'Enter' || ev.key == ' ') {
+      viewGrid.toggleTag(personId, column.config.tag_id, !data);
+      ev.defaultMuiPrevented = true;
+    }
+  }
+
+  processRowUpdate(
+    viewGrid: UseViewGridReturn,
+    col: PersonTagViewColumn,
+    personId: number,
+    data: string | undefined
+  ): void {
+    viewGrid.toggleTag(personId, col.config.tag_id, !!data, data);
+  }
+}
+
+const useStyles = makeStyles(() => ({
+  ghost: {
+    pointerEvents: 'none',
+  },
+  ghostContainer: {
+    '&:hover': {
+      opacity: 0.4,
+    },
+    cursor: 'pointer',
+    opacity: 0,
+    transition: 'opacity 0.1s',
+  },
+  valueTagEditCell: {},
+}));
+
+interface CellProps {
+  cellValue: ZetkinTag | undefined;
+  personId: number;
+  tag: ZetkinTag | null;
+}
+
+const Cell: FC<CellProps> = ({ cellValue, personId, tag }) => {
+  if (!tag) {
+    return null;
```

Comment:
This breaks "restricted view", because a user with which a list has been shared, who is not an official, won't be able to load the tags standalone. So the cells render nothing.

In restricted mode, the list will never be editable, so just rendering the cells where a tag exists should be fine, and getting the value type from the tag in the row should be fine. See my next comment.

![image](https://github.com/zetkin/app.zetkin.org/assets/550212/bd083b67-d91d-4165-9378-947faf041730)

Reply:
I completely forgot to test this usecase!! Thank you for catching it! :D